Adding DateFormat augmentation

motiwari opened the pull request:
This changes the forms of dates and times that can appear in sentences, e.g. "5 p.m." to "1700"

motiwari commented:
Hi, this PR is still a work-in-progress; please do not review

rteehas commented:
> Hi, this PR is still a work-in-progress; please do not review

Is this still in a do not review stage? The deadline for me to submit reviews is the 18th I believe

motiwari commented:
Thanks @rteehas , still a WIP, will update shortly!

rteehas commented:
Tagging @kaustubhdhole to see if I can hold off on reviewing this until it is in its final state

motiwari commented:
> Tagging @kaustubhdhole to see if I can hold off on reviewing this until it is in its final state

Thanks -- almost done!

motiwari commented:
Hey @rteehas done! Apologies for the delay.

Please see the updated code, PR title, and PR comments. Let me know if you have any questions!

Perhaps surprisingly, this transformation was rather nontrivial. I hope it will add value as an augmentation!

ashish3586 commented on transformations/wordstonumbers/transformation.py:
```diff
@@ -0,0 +1,18 @@
+
+
+class WordsToNumbers(SentenceOperation):
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION, TaskType.TEXT_TAGGING]
```
Task TEXT_TAGGING is not applicable here because of a change in the number of words (Ex: I have two hundred fifty books --> I have 250 books.)

ashish3586 replied:
You can add tasks PARAPHRASE_DETECTION, TEXTUAL_ENTAILMENT

ashish3586 commented on transformations/wordstonumbers/transformation.py:
```diff
@@ -0,0 +1,18 @@
+
+from text2nums import *
+
+
```
Please add a docstring to the class WordToNumbers.

ashish3586 commented on transformations/wordstonumbers/README.md:
```diff
@@ -0,0 +1,38 @@
+## What tasks does it intend to benefit?
+This perturbation would benefit all tasks which have a sentence/paragraph/document as input like text classification, 
+text generation, etc. and may deal with numbers written out in word form. 
+
+## Previous Work
+
+Several webpages exist to do this (as the code is fairly simple) but have various errors:
+
+- https://www.browserling.com/tools/words-to-numbers cannot handle capital letters
+- https://www.dcode.fr/writing-words-numbers does not provide source code
+
+Our code is very loosely adapted from
+https://stackoverflow.com/questions/493174/is-there-a-way-to-convert-number-words-to-integers, though our implementation
+is more general and handles sentences where only part of the sentence refers to a number.
+
+## What are the limitations of this transformation?
```
Please add a robustness evaluation section here like PR #218.

kaustubhdhole commented on transformations/wordstonumbers/README.md:
```diff
@@ -0,0 +1,38 @@
+Our code is very loosely adapted from
+https://stackoverflow.com/questions/493174/is-there-a-way-to-convert-number-words-to-integers, though our implementation
+is more general and handles sentences where only part of the sentence refers to a number.
+
```
Also, you might want to add this to the README and mention a line or two how your transformation is different. https://github.com/GEM-benchmark/NL-Augmenter/blob/main/transformations/number-to-word/transformation.py

motiwari commented:
Hi @kaustubhdhole and @ashish3586 , thanks for the comments above -- I've addressed all your comments except adding the robustness evaluation.

I just rebased this branch on `main` and am now getting the following error that don't look related to my code; do you know how to fix? It's for the robustness evaluation:

```
(base) ➜  NL-Augmenter git:(dateformat) ✗ python evaluate.py -t WordsToNumbers -task "TEXT_CLASSIFICATION" -m "textattack/roberta-base-imdb" -d "imdb" -p 20
Traceback (most recent call last):
  File "evaluate.py", line 3, in <module>
    from evaluation.evaluation_engine import evaluate
  File "/Users/motiwari/Desktop/NL-Augmenter/evaluation/evaluation_engine.py", line 1, in <module>
    from evaluation import (
  File "/Users/motiwari/Desktop/NL-Augmenter/evaluation/evaluate_ner_tagging.py", line 2, in <module>
    from datasets import load_dataset
ModuleNotFoundError: No module named 'datasets'
```

motiwari commented:
@kaustubhdhole @ashish3586 @rteehas @sebastianGehrmann can you take a look at this?
